redbot/cogs/alias/alias_entry.py

```diff
@@ -30,34 +47,195 @@ def inc(self):
         self.uses += 1
         return self.uses
 
+    def get_extra_args_from_alias(self, message: discord.Message, prefix: str) -> str:
+        """
+        When an alias is executed by a user in chat this function tries
+            to get any extra arguments passed in with the call.
+            Whitespace will be trimmed from both ends.
+        :param message:
+        :param prefix:
+        :param alias:
+        :return:
+        """
+        known_content_length = len(prefix) + len(self.name)
+        extra = message.content[known_content_length:]
+        view = StringView(extra)
+        view.skip_ws()
+        extra = []
+        while not view.eof:
+            prev = view.index
+            word = view.get_quoted_word()
+            if len(word) < view.index - prev:
+                word = "".join((view.buffer[prev], word, view.buffer[view.index - 1]))
+            extra.append(word)
+            view.skip_ws()
+        return extra
+
     def to_json(self) -> dict:
-        try:
-            creator = str(self.creator.id)
-            guild = str(self.guild.id)
-        except AttributeError:
-            creator = self.creator
-            guild = self.guild
 
         return {
             "name": self.name,
             "command": self.command,
-            "creator": creator,
-            "guild": guild,
-            "global": self.global_,
+            "creator": self.creator,
+            "guild": self.guild,
             "uses": self.uses,
         }
 
     @classmethod
     def from_json(cls, data: dict, bot: commands.Bot = None):
-        ret = cls(data["name"], data["command"], data["creator"], global_=data["global"])
+        ret = cls(data["name"], data["command"], data["creator"], data["guild"])
 
         if bot:
+            # this might become useful later but as of right now it does nothing
             ret.has_real_data = True
             ret.creator = bot.get_user(int(data["creator"]))
             guild = bot.get_guild(int(data["guild"]))
             ret.guild = guild
-        else:
-            ret.guild = data["guild"]
 
         ret.uses = data.get("uses", 0)
         return ret
+
+
+class AliasCache:
+    def __init__(self, config: Config, cache_enabled: bool = True):
+        self.config = config
+        self._cache_enabled = cache_enabled
+        self._loaded = False
+        self._aliases: Dict[Optional[int], Dict[str, AliasEntry]] = {None: {}}
+
+    async def load_aliases(self):
+        for alias in await self.config.entries():
+            self._aliases[None][alias["name"]] = AliasEntry.from_json(alias)
+
+        for guild_id in await self.config.all_guilds():
+            aliases = await self.config.guild_from_id(guild_id).entries()
```

Comment:
You shouldn't be calling Config on each guild again if you already have all the needed data.

Also, considering all the optimizations for big bots that we've done lately, this should probably be wrapped in `AsyncIter` class from `redbot.core.utils` to avoid blocking event loop on alias loading for bigger bots.

```suggestion
        all_guilds = await self.config.all_guilds()
        async for guild_id, guild_data in AsyncIter(all_guilds.items(), steps=100):
            aliases = guild_data["entries"]
```